src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -182,6 +182,19 @@ public string HubName
         /// </summary>
         public bool UseGracefulShutdown { get; set; } = false;
 
+        /// <summary>
+        /// Controls whether an uncaught exception inside an entity operation should roll back the effects of the operation.
+        /// </summary>
+        /// <remarks>
+        /// The rollback undoes all internal effects of an operation
+        /// (sent signals, and state creation, deletion, or modification).
+        /// However, it does not roll back external effects (such as I/O that was performed).
+        /// This setting can affect serialization overhead: if true, the entity state is serialized
+        /// after each individual operation. If false, the entity state is serialized
+        /// only after an entire batch of operations completes.
+        /// </remarks>
+        public bool RollbackEntityStateOnExceptions { get; set; } = true;
```

Comment:
can we change this to false for `2.2.0`?

Reply:
yes, will do that.

Reply:
I am having second thoughts ... the behavior of our current implementation (without rollback) is kind of buggy: I just found a bug as I was adding tests like you suggested, this bug means that serialization errors were not properly reported in an operation

So enabling this by default could help remove some user confusion and leave us with a better thought out and tested design. It's not just about the state rollback... it's also about reporting serialization exceptions promptly for an operation, as opposed to only after the entire batch is done.

I think keeping this to true as default provides overall better experience for our users - I would hate to get more issues asking about weird behavior that we already fixed.

Reply:
Are we sure that the bug was present before this PR, and not caused by this refactor? Your refactor seems to keep the old path pretty much intact, but it's hard to say without testing. 

In general, if you think this is safe enough to turn on as the new default behavior in 2.2.0, I trust your stance. I just want to make sure we have good tests for both paths if at all possible.

Reply:
Yes, the bug was there in the old behavior, before this PR. There were not enough tests for the exception behavior. I have added tests now for both the old and new behavior. But I don't see much point in keeping the old behavior as a default (it is also not documented).